core/rpc/src/author/mod.rs

```diff
@@ -176,4 +188,24 @@ impl<B, E, P, RA> AuthorApi<ExHash<P>, BlockHash<P>> for Author<B, E, P, RA> whe
 	fn unwatch_extrinsic(&self, _metadata: Option<Self::Metadata>, id: SubscriptionId) -> Result<bool> {
 		Ok(self.subscriptions.cancel(id))
 	}
+
+	fn set_key(&self, key_type_id: KeyTypeId, key: Vec<u8>) -> Result<()> {
```

Comment:
doing the keygen in this function is almost definitely wrong. The `key` bytes should be the key itself, not a seed. then you wouldn't need to try and do an exhaustive match on the key-type-id.

Reply:
How would this avoid doing an exhaustive match? We need to match on KeyTypeId to go from Vec<u8> to impl Pair.